### PR TITLE
feat(core): Switch core client in v6provider to use generated `next` client

### PR DESCRIPTION
**Pull request by ikadix**
Uses the new [go-katapult/next/core](https://github.com/krystal/go-katapult/tree/main/next/core) package in the v6provider.

### CODE REVIEW

**Review by jimehk — CHANGES_REQUESTED** [review]

File: internal/v6provider/meta.go [hunk 1]
```diff
@@ -117,6 +115,12 @@ func NewMeta(
 
 	opts = append(opts, katapult.WithHTTPClient(httpClient))
 
+	serverURL := &url.URL{
+		Scheme: "https",
+		Host:   "api.katapult.io",
+		Path:   "/core/v1",
+	}
```
[the remark this answers is not in the document]
Comment (ikadix) [reply internal/v6provider/meta.go:122]: It does as of the most recent update I believe

[the remark this answers is not in the document]
Comment (ikadix) [reply internal/v6provider/meta.go:122]: The schema contains one but the client still requires this it seems

File: internal/v6provider/resource_load_balancer_rule.go [hunk 2]
```diff
@@ -522,42 +570,56 @@ func (r *LoadBalancerRuleResource) LoadBalancerRuleRead(
...
+
+	lbr := res.JSON200.LoadBalancerRule
+
+	model.ID = types.StringPointerValue(lbr.Id)
+	model.LoadBalancerID = types.StringPointerValue(lbr.LoadBalancer.Id)
+	model.Algorithm = types.StringValue(string(*lbr.Algorithm))
+	model.DestinationPort = types.Int64Value(int64(*lbr.DestinationPort))
+	model.ListenPort = types.Int64Value(int64(*lbr.ListenPort))
+	model.Protocol = types.StringValue(string(*lbr.Protocol))
+	model.ProxyProtocol = types.BoolValue(*lbr.ProxyProtocol)
```
[the remark this answers is not in the document]
Comment (ikadix) [reply internal/v6provider/resource_load_balancer_rule.go:598]: I am assuming you mean `BoolPointerValue` ?

File: internal/v6provider/resource_ip.go [hunk 3]
```diff
@@ -175,46 +175,61 @@ func (r *IPResource) Create(
...
+			})
 		if err != nil {
 			resp.Diagnostics.AddError(
 				"Default Network Error",
 				err.Error(),
 			)
 			return
 		}
+
+		networkID = res.JSON200.Network.Id
```
[the remark this answers is not in the document]
Comment (ikadix) [reply internal/v6provider/resource_ip.go:195]: Yeah, for reference with that change in the go-katapult to return an error on non 2xx responses, this should be fixed :shrug:  but handling without that is fine :pray:

File: internal/v6provider/testdata/LoadBalancerRule_minimal.cassette.yaml [hunk 4]
```diff
@@ -29,38 +26,36 @@ interactions:
...
 - request:
     body: ""
     form: {}
     headers:
-      Accept:
-      - application/json
       User-Agent:
-      - Terraform/ (+https://www.terraform.io) Terraform-Plugin-Framework terraform-provider-katapult
-    url: https://api.katapult.io/core/v1/load_balancers/_?load_balancer%5Bid%5D=lb_kwRBghlBEWmcRwaf
+      - go-katapult/0.2.0
```
[the remark this answers is not in the document]
Comment (ikadix) [reply internal/v6provider/testdata/LoadBalancerRule_minimal.cassette.yaml:52]: We can throw in a request editor func yeah

File: internal/v6provider/resource_load_balancer.go [hunk 5]
```diff
@@ -261,31 +284,46 @@ func (r *LoadBalancerResource) Update(
...
 		!plan.TagIDs.Equal(state.TagIDs) {
 		t, ids := extractLoadBalancerResourceTypeAndIDs(&plan)
-		args.ResourceType = t
-		args.ResourceIDs = &ids
+		args.Properties.ResourceType = &t
+		args.Properties.ResourceIds = &ids
+	} else {
+		t, ids := extractLoadBalancerResourceTypeAndIDs(&state)
+		args.Properties.ResourceType = &t
+		args.Properties.ResourceIds = &ids
```
[the remark this answers is not in the document]
Comment (ikadix) [reply internal/v6provider/resource_load_balancer.go]: oh I don't believe so actually

File: internal/v6provider/resource_load_balancer_rule.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jimehk) [reply internal/v6provider/resource_load_balancer_rule.go]: I very much do, yes 🤦‍♂️... lol

**Review by jimehk — APPROVED** [review]

**Pull request merged**